docs/build-docs

```diff
@@ -0,0 +1,144 @@
+    shutil.copy(DOCS_DIR / "conf.py", conf_copy)
+    shutil.copy(DOCS_DIR / "CQCLogo.png", logo_copy)
+    remove_dir(mod_build)
+    subprocess.run(
+        [
+            "sphinx-build",
+            "-b",
+            "html",
+            "-D",
+            f"project=pytket-{module}",
+            "-D",
+            f"copyright={datetime.date.today().year} Cambridge Quantum Computing",
+            "-D",
+            f"version={'.'.join(v[:2])}",
+            "-D",
+            f"release={'.'.join(v)}",
+            ".",
+            "build",
+        ],
+        cwd=mod_docs,
+    )
+    for htmlfile in mod_build.rglob("*.html"):
+        fix_links(htmlfile)
+    fix_links(mod_build / "searchindex.js")
+    conf_copy.unlink()
+    logo_copy.unlink()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Build HTML documentation for one or more modules."
+    )
+    parser.add_argument(
+        "-m",
+        "--modules",
+        nargs="*",
+        default=[],
+        help="names of modules to build (without the `pytket-` prefix)",
+    )
+    parser.add_argument("-a", "--all", action="store_true", help="build all modules")
+    parser.add_argument("-i", "--index", action="store_true", help="build index page")
+    parser.add_argument("-d", "--dest", help="copy artifacts into destination folder")
+    args = parser.parse_args()
+
+    if args.all:
+        modules_to_build = get_all_modules()
+    else:
+        modules_to_build = args.modules
+
+    print("Building docs for modules:", modules_to_build)
+    for module in modules_to_build:
+        build_module_docs(module)
+
+    if args.index:
+        print("Building index page...")
+        index_rst = DOCS_DIR / "index.rst"
+        with open(DOCS_DIR / "intro.txt", "r") as f:
+            content = f.readlines()
+            content.append("\n")
+            for module in get_all_modules():
+                content.append(f"- `pytket-{module} <{module}/index.html>`_\n")
+        with open(index_rst, "w") as f:
+            f.writelines(content)
+        remove_dir(DOCS_DIR / "build")
+        subprocess.run(
+            [
+                "sphinx-build",
+                "-b",
+                "html",
+                "-D",
+                f"project=pytket-extensions",
+                "-D",
+                f"copyright={datetime.date.today().year} Cambridge Quantum Computing",
+                "-D",
+                "templates_path=templates",
+                ".",
+                "build",
+            ],
+            cwd=DOCS_DIR,
+        )
+        index_rst.unlink()
+
+    if args.dest is not None:
+        dest = Path(args.dest)
+        for module in modules_to_build:
+            shutil.copytree(
+                MODULES_DIR / f"pytket-{module}" / "docs" / "build",
+                dest / module,
+                dirs_exist_ok=True,
+            )
+        if args.index:
+            for f in (DOCS_DIR / "build").iterdir():
```

Comment:
```
shutil.copytree(
    DOCS_DIR / "build",
    dest,
    dirs_exist_ok=True,
)
```
wouldn't this have the same effect?

Reply:
The difference, I think, is that that would replace `dest`, so that any files existing in it would be deleted. I wanted to be able to keep existing files and only overwrite ones with matching names.